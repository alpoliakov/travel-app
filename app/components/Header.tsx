import { motion } from 'framer-motion';
import Image from 'next/image';
import Link from 'next/link';
import { useRouter } from 'next/router';
import setLanguage from 'next-translate/setLanguage';
import React, { useEffect, useState } from 'react';
import toast from 'react-hot-toast';
import { useIntl } from 'react-intl';

import { useAuth } from '../lib/useAuth';
import Dropdown from './Dropdown';
import Search from './Search';

const postVariants = {
  initial: { scale: 0.96, y: 30, opacity: 0 },
  enter: {
    scale: 1,
    y: 0,
    opacity: 1,
    transition: { duration: 0.5, ease: [0.48, 0.15, 0.25, 0.96] },
  },
  exit: {
    scale: 0.6,
    y: 100,
    opacity: 0,
    transition: { duration: 0.2, ease: [0.48, 0.15, 0.25, 0.96] },
  },
};

export default function Header() {
  const { locale, locales } = useRouter();
  const [currentLocal, setCurrentLocal] = useState(locale);
  const { formatMessage: f } = useIntl();
  const router = useRouter();
  const { user, signOut, message, error } = useAuth();

  const showSearch = router.pathname === '/';

  const setLocal = ({ target }) => {
    setCurrentLocal(target.value);
  };

  useEffect(() => {
    setLanguage(currentLocal);
  }, [currentLocal]);

  useEffect(() => {
    if (message) {
      toast.success(message, { duration: 4000 });
    }
  }, [message]);

  useEffect(() => {
    if (error) {
      toast.error(error, { duration: 4000 });
    }
  }, [error]);

  return (
    <div className="header pb-5">
      <div className="header__home">
        <Link href="/">
          <a>
            <Image src="/images/logo.png" width="150px" height="80px" />
          </a>
        </Link>
      </div>
      <div className="header__search">
        {showSearch && (
          <motion.h1 initial="initial" animate="enter" exit="exit" variants={postVariants}>
            <Search />
          </motion.h1>
        )}
      </div>
      <div className="header__menu">
        <select className="select-css" onChange={setLocal} defaultValue={locale}>
          {locales.map((item) => (
            <option key={item} value={item}>
              {item}
            </option>
          ))}
        </select>
        {user && <Dropdown />}
        {!user && (
          <>
            <Link href="/auth/login">
              <a className="text-white p-1 hover:bg-opacity-100 hover:text-opacity-70 text-base p-2 bg-gray-900 rounded-xl bg-opacity-50">
                {f({ id: 'btnLogin' })}
              </a>
            </Link>
            <Link href="/auth/signup">
              <a className="text-white p-1 hover:bg-opacity-100 hover:text-opacity-70 text-base p-2 bg-gray-900 rounded-xl bg-opacity-50">
                {f({ id: 'btnSignUp' })}
              </a>
            </Link>
          </>
        )}
      </div>
    </div>
  );
}
