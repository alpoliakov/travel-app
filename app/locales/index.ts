export const messages = {
  en: {
    btnLogin: 'Login',
    btnSignUp: 'Sign Up',
    search: 'Search',
    phEmail: 'Enter your e-mail',
    phPass: 'Enter your password',
    createAcc: 'Create an account',
    forgetPass: 'Forgot password ?',
    enter: 'Enter',
    load: 'Loading',
    high: 'Max',
    low: 'Min',
    now: 'Currently',
    pass: 'Password (6 characters minimum):',
    phName: 'Enter your name',
    phAvatar: 'Upload your avatar',
    name: 'Name (3 characters minimum):',
    avatar: 'Avatar:',
    register: 'Sign Up',
    signOut: 'Sign Out',
    errorEmail: 'Invalid email field!',
    errorPass: 'Invalid password field!',
    errorName: 'Invalid username field!',
    profile: 'Profile',
    delete: 'Delete',
    or: 'Or',
    selectFile: 'Select a file',
    hello: 'Hello!',
    modalText: 'To view, you need to log in or register!',
    close: 'Close',
  },
  ru: {
    btnLogin: 'Войти',
    btnSignUp: 'Регистрация',
    search: 'Поиск',
    phEmail: 'Введите ваш e-mail',
    phPass: 'Введите ваш пароль',
    createAcc: 'Создать аккаунт',
    forgetPass: 'Забыли пароль ?',
    enter: 'Ввести',
    load: 'Загрузка',
    high: 'Макс',
    low: 'Мин',
    now: 'Сейчас',
    pass: 'Пароль (6 символов минимум):',
    phName: 'Введите ваше имя',
    phAvatar: 'Загрузите ваш аватар',
    name: 'Имя (3 символа минимум):',
    avatar: 'Аватар:',
    register: 'Регистрация',
    signOut: 'Выйти',
    errorEmail: 'Невалидное поле электронной почты!',
    errorPass: 'Невалидное поле пароля!',
    errorName: 'Невалидное поле имени пользователя!',
    profile: 'Профиль',
    delete: 'Удалить',
    or: 'Или',
    selectFile: 'Выберите файл',
    hello: 'Приветствуем!',
    modalText: 'Для просмотра необходимо авторизоваться или зарегистрироваться!',
    close: 'Закрыть',
  },
  uk: {
    btnLogin: 'Увійти',
    btnSignUp: 'Реєстрація',
    search: 'Пошук',
    phEmail: 'Введіть ваш e-mail',
    phPass: 'Введіть ваш пароль',
    createAcc: 'Створити акаунт',
    forgetPass: 'Забули пароль ?',
    enter: 'Ввести',
    load: 'Завантаження',
    high: 'Макс',
    low: 'Miн',
    now: 'Зараз',
    pass: 'Пароль (6 символів мінімум):',
    phName: 'Введіть ваше им`я',
    phAvatar: 'Завантажте ваш аватар',
    name: 'Ім`я (3 символа мінімум):',
    avatar: 'Аватар:',
    register: 'Реєстрація',
    signOut: 'Вийти',
    errorEmail: 'Невалідне поле електронної пошти!',
    errorPass: 'Невалідне поле пароля!',
    errorName: 'Невалідне поле імені користувача',
    profile: 'Профіль',
    delete: 'Видалити',
    or: 'Або',
    selectFile: 'Виберіть файл',
    hello: 'Вітаємо!',
    modalText: 'Для перегляду потрібно увійти або зареєструватися!',
    close: 'Закрити',
  },
};
