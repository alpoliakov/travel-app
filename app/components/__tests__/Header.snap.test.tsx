import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';

import { messages } from '../../locales';
import Header from '../Header';

jest.mock('../../lib/useAuth', () => ({
  useAuth: jest.fn(() => ({ signIn: jest.fn(), user: '', error: '' })),
  AuthProvider: ({ children }) => <div>{children}</div>,
}));

jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: () => ({
    route: '/',
    pathname: '',
    query: '',
    asPath: '',
    locale: 'en',
    locales: ['en', 'ru', 'uk'],
  }),
}));

it('Header renders correctly', () => {
  const tree = renderer
    .create(
      <IntlProvider locale="ru" messages={messages['ru']}>
        <Header />)
      </IntlProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
