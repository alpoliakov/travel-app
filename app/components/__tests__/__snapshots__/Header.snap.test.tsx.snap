// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
Array [
  <div
    className="header pb-5"
  >
    <div
      className="header__home"
    >
      <a
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "margin": 0,
              "maxWidth": "100%",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "maxWidth": "100%",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUwIiBoZWlnaHQ9IjgwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
              style={
                Object {
                  "border": "none",
                  "display": "block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "padding": 0,
                }
              }
            />
          </div>
          <img
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "visibility": "hidden",
                "width": 0,
              }
            }
          />
        </div>
      </a>
    </div>
    <div
      className="header__search"
    />
    <div
      className="header__menu"
    >
      <select
        className="select-css"
        data-testid="select-locale"
        defaultValue="en"
        onChange={[Function]}
      >
        <option
          value="en"
        >
          en
        </option>
        <option
          value="ru"
        >
          ru
        </option>
        <option
          value="uk"
        >
          uk
        </option>
      </select>
      
      <a
        className="text-white p-1 hover:bg-opacity-100 hover:text-opacity-70 text-base p-2 bg-gray-900 rounded-xl bg-opacity-50"
        href="/auth/login"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Войти
      </a>
      <a
        className="text-white p-1 hover:bg-opacity-100 hover:text-opacity-70 text-base p-2 bg-gray-900 rounded-xl bg-opacity-50"
        href="/auth/signup"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Регистрация
      </a>
    </div>
  </div>,
  ")",
]
`;
